File: LeetCode/Python/Medium/2149-Rearrange-Array-Elements-by-Sign/solution.py
from collections import deque

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        result=list()

        positives=deque([number for number in nums if number > 0])

        negatives=deque([number for number in nums if number < 0])

        while positives or negatives:
            if positives:
                result.append(positives.popleft())

            if negatives:
                result.append(negatives.popleft())

        return result

File: LeetCode/Rust/Medium/2149-Rearrange-Array-Elements-by-Sign/solution.rs
use std::collections::{VecDeque};

impl Solution
{
    pub fn rearrange_array(nums: Vec<i32>) -> Vec<i32>
    {
        let mut result: Vec<i32>=Vec::new();

        let mut negatives: VecDeque<i32>=VecDeque::new();

        let mut positives: VecDeque<i32>=VecDeque::new();

        for number in nums.iter()
        {
            match number
            {
                number if *number > 0 =>
                {
                    positives.push_back(*number);
                },
                number if *number < 0 =>
                {
                    negatives.push_back(*number);
                },
                _ => ()
            }
        }

        while let (Some(positive), Some(negative))=(positives.pop_front(), negatives.pop_front())
        {
            result.push(positive);

            result.push(negative);
        }

        result
    }
}